### ISSUE TITLE
The previously found element \"Application 'XXX'\" is not present in the current view anymore.

### ISSUE THREAD

**Issue by Xuuxxxx** [appium/appium#19297]
**Env**: 
Appium 2, latest Appium Inspector from https://github.com/appium/appium-inspector
**capability**:
<img width="510" alt="image" src="https://github.com/appium/appium/assets/49302588/9addff55-5230-4b2a-8b51-d6d7236a9d73">

**Situation**:
Ps: My app is not a native ios app, it is done by react-native.
I am doing UI testing, before I login my app I can see the page in Appium Inspector. Every page before I log in, I can click refresh button to load. But after I have logged in, when I click refresh button of the Appium inspector, the page will not refreshed. 
<img width="1081" alt="image" src="https://github.com/appium/appium/assets/49302588/db2f5c46-dc38-4a04-8dfa-85865a01db01">
**Appium** **log**:
<img width="983" alt="image" src="https://github.com/appium/appium/assets/49302588/6e70a4d6-7c3c-40bb-bf22-e5db09758634">
<img width="978" alt="image" src="https://github.com/appium/appium/assets/49302588/c9daf635-be74-43c8-b7b6-5c3756b753c7">
<img width="980" alt="image" src="https://github.com/appium/appium/assets/49302588/8042ac46-53f7-4e02-9347-08142e1ce6b6">

**Comment by KazuCocoa** [comment]
Could you attache the full appium log as GIST as well?

**Comment by mykola-mokhnach** [comment]
I was observing similar errors when the app under test had too large page source. In such case XCTest simply fails to serialise it within sane time interval.
You could probably reduce the value of snapshotMaxDepth as a possible workaround. See https://github.com/appium/appium/issues/14825 for more details.

**Comment by Xuuxxxx** [comment]
@KazuCocoa Sorry for my late. This is the first time I asked issue, I will attach the full log as  GIST in the future.

@mykola-mokhnach I have refer to your linked issue.  I tried to reduce the snapshotMaxDepth to 62, and it works now,  I can load the page now. But since the max depth is 50, my element depth >50, so I can not select the specific element in the page.  
But when I set the snapshotMaxDepth to 63, I can not load the page.
Can any capability help to solve this issue?
